src/java.base/share/classes/jdk/internal/jimage/BasicImageReader.java

```diff
@@ -103,30 +106,8 @@ protected BasicImageReader(Path path, ByteOrder byteOrder)
         if (map != null && MAP_ALL) {
             channel = null;
         } else {
-            channel = FileChannel.open(imagePath, StandardOpenOption.READ);
-            // No lambdas during bootstrap
-            AccessController.doPrivileged(new PrivilegedAction<Void>() {
-                @Override
-                public Void run() {
-                    if (BasicImageReader.class.getClassLoader() == null) {
-                        try {
-                            Class<?> fileChannelImpl =
-                                Class.forName("sun.nio.ch.FileChannelImpl");
-                            Method setUninterruptible =
-                                    fileChannelImpl.getMethod("setUninterruptible");
-                            setUninterruptible.invoke(channel);
-                        } catch (ClassNotFoundException |
-                                 NoSuchMethodException |
-                                 IllegalAccessException |
-                                 InvocationTargetException ex) {
-                            // fall thru - will only happen on JDK-8 systems where this code
-                            // is only used by tools using jrt-fs (non-critical.)
-                        }
-                    }
-
-                    return null;
-                }
-            });
+            channel = openFileChannel();
```

Comment:
Could you avoid this refactoring to simplify future merges?

Reply:
Do you mean to just copy-paste the `openFileChannel()` code? Do we rather want any change in that code go unnoticed?

Reply:
Oh! I see, you did refactoring to call openFileChannel() from registerIfCRaCPresent(). OK, that makes sense.